mlxtend/feature_selection/sequential_feature_selector.py

```diff
@@ -331,20 +392,26 @@ def fit(self, X, y, custom_feature_names=None, groups=None, **fit_params):
         self.k_score_ = None
 
         self.fixed_features_ = self.fixed_features
-        self.fixed_features_set_ = set()
-
         if hasattr(X, "loc"):
             X_ = X.values
-            if self.fixed_features is not None:
-                self.fixed_features_ = tuple(
-                    X.columns.get_loc(c) if isinstance(c, str) else c
-                    for c in self.fixed_features
-                )
+            self.fixed_features_ = tuple(
+                X.columns.get_loc(c) if isinstance(c, str) else c
+                for c in self.fixed_features_
+            )
         else:
             X_ = X
+        self.fixed_features_set_ = set(self.fixed_features_)
+
+        if self.feature_groups is None:
+            self.feature_groups = [[i] for i in range(X_.shape[1])]
 
-        if self.fixed_features is not None:
-            self.fixed_features_set_ = set(self.fixed_features_)
+        features_groupID = np.full(X_.shape[1], -1, dtype=np.int64)
+        for id, group in enumerate(self.feature_groups):
```

Comment:
`id` is a built-in function in python. Please replace it with `group_id`

Reply:
fixed it!

Reply:
Oh.. thanks! that was a comment for myself so that I do not miss it later. Thanks for taking care of it!!!